src/components/Form/ToggleButton.tsx

```diff
@@ -0,0 +1,57 @@
+import React, { Component } from 'react';
+import { HtmlButton } from '../../reset';
+import { ToggleProps, baseClassName, ToggleState } from './Toggle';
+import { logger } from '../../utils/logger';
+
+export class ToggleButton extends Component<ToggleProps> {
+  state: ToggleState = {
+    toggleState: !!this.props.checked || !!this.props.defaultChecked || false,
+  };
+
+  handleClick = () => {
+    const { checked, onClick } = this.props;
+    const { toggleState } = this.state;
+    if (checked === undefined) {
+      this.setState({ toggleState: !toggleState });
+    }
+    if (!!onClick) {
+      onClick({ toggleState: !toggleState });
+    }
+  };
+
+  render() {
+    const {
+      disabled = false,
+      children,
+      checked: dissMissChecked,
+      defaultChecked: dissMissDefaultChecked,
+      onClick: dissMissOnClick,
+      toggleInputProps,
+      toggleInputComponent,
+      ...passProps
+    } = this.props;
+    const { toggleState } = this.state;
+    const toggleButtonClassName = `${baseClassName}_button`;
+
+    if (toggleInputProps || toggleInputComponent) {
```

Comment:
I would use double negative (`!!toggleInputProps || !!toggleInputComponent`) checking here - also because these props come from outside the library.

Reply:
Changed to use double negation.